### ISSUE TITLE
MCW V14

### ISSUE THREAD

**Issue by jamesmfriedman** [rmwc/rmwc#799]
Master Task for bringing all MCW components up to current. All components should also include updated tests using @react/testing-library

Tasks
- [ ] Convert `tag` prop to `as` and look into using a polymorphic helper instead of custom createComponent
- [ ] Discovery around useFoundation api cleanup
- [x] Convert to @react/testing-library and remove Enzyme
- [ ] Fix storybook (currently loading infinitely) 

Components
- [x] button (https://github.com/rmwc/rmwc/pull/816)
- [x] card (https://github.com/rmwc/rmwc/pull/818)
- [x] checkbox (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] chips (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] circular-progress (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] data-table (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] density
- [x] dialog (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] drawer (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] elevation (https://github.com/rmwc/rmwc/pull/822)
- [x] fab (https://github.com/rmwc/rmwc/pull/821)
- [x] feature-targeting
- [x] floating-label (https://github.com/rmwc/rmwc/pull/822)
- [ ] focus-ring (not ready, see https://github.com/material-components/material-components-web/tree/v14.0.0/packages/mdc-focus-ring)
- [x] form-field (https://github.com/rmwc/rmwc/pull/819)
- [x] icon-button (https://github.com/rmwc/rmwc/pull/822)
- [x] image-list (https://github.com/rmwc/rmwc/pull/822)
- [x] layout-grid (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] line-ripple (https://github.com/rmwc/rmwc/pull/822)
- [x] linear-progress (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] list (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] menu-surface (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] menu (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] notched-outline (https://github.com/rmwc/rmwc/pull/822)
- [x] progress-indicator (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] radio (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] ripple (https://github.com/rmwc/rmwc/pull/822)
- [x] segmented-button (on https://github.com/rmwc/rmwc/pull/907)
- [x] select (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [ ] shape
- [x] slider (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] snackbar (https://github.com/rmwc/rmwc/pull/823)
- [x] switch (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] tabs (on https://github.com/rmwc/rmwc/tree/v14/all-components)
  - [x] tab-bar
  - [x] tab-indicator
  - [x] tab-scroller
  - [x] tab
- [x] textfield (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] theme (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [ ] tokens (not ready see https://github.com/material-components/material-components-web/tree/v14.0.0/packages/mdc-tokens)
- [x] tooltip (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] top-app-bar (on https://github.com/rmwc/rmwc/tree/v14/all-components)
- [x] touch-target (https://github.com/rmwc/rmwc/pull/816)
- [x] typography (https://github.com/rmwc/rmwc/pull/820)

- [x] Get branch running again @b-d-m-p @EmiBemi 
- [x] Merge v14 into next and make a beta release branch
- [x] add rctooltip for backwards compatibility (people will have to change their imports, add in documentation) @jensborch 
- [x] add useFoundation to new tooltip
- [x] remove remining links etc. to Open Collective
- [ ] Merge new miami theme (merge to next) @b-d-m-p 
- [ ] Add Sass compiler @b-d-m-p 
- [x] look for bugs and test @b-d-m-p 
- [x] Fix issues in the beta @EmiBemi 
- [x] Get all team members to test
- [x] Release the branch to the community and ask for feedback

**Comment by b-d-m-p** [comment]
When running the docs locally on this branch or one of the v14 component branches and you get an error like this:
```
ERROR in ./src/avatar/index.tsx 1:40-132

Module not found: Error: You attempted to import /Users/you/Documents/GitHub/rmwc/node_modules/react-refresh/runtime.js which falls outside of the project src/ directory. Relative imports outside of src/ are not supported.
You can either move it inside src/, or add a symlink to it from project's node_modules/.
```
... then run `npm dedupe` and `npm install` and it will fix it. Thanks to @EmiBemi .

**Comment by b-d-m-p** [comment]
@b-d-m-p check out why this is failing

**Comment by b-d-m-p** [comment]
Emi merged this into the next branch, we want to share that with the community. 

- [ ] We need to bump the version (in the next branch) @EmiBemi @jensborch 
- [ ] we need to do a prerelease @EmiBemi 
- [ ] Get all the members to test it @b-d-m-p 
- [ ] Share it to the community @b-d-m-p

**Comment by b-d-m-p** [comment]
Updates:
There was an issue with github actions, so @jensborch is fixing this. These tasks are remaining.

- [ ]  we need to do a prerelease @EmiBemi
- [ ]  Get all the members to test it @b-d-m-p
- [ ]  Share it to the community @b-d-m-p

**Comment by b-d-m-p** [comment]
We had a meeting and talked about what is blocking the release and figured out the issue.

What we had done:
- We created a PR and merged `release/major-2023-07-10` #977 into `next` to release it as a pre-release, but it did not release via the Github Action, which we expected it to do.

What we figured out:
- We looked at the Github Actions and noticed in the Publish Action that there was [ci(rmwc): release of 8.0.8](https://github.com/rmwc/rmwc/actions/runs/5209253776) for the last release but not one for v14. We looked in the publish.yml file and saw that the action is triggered on merging into the `master` branch only, not the `next` branch which is used for pre-releases. We determined that this is the issue that prevented the pre-release from being created. 

What our questions are:
- If the Publish Action only works for major releases and not pre-releases (because it only triggers on a PR merge into `master` and not `next`), how did James create these pre-releases before? Manually?

What actions we should/could take:
- We should ask James first of all.
- Just thinking about the issue, we could edit the `publish.yml` file to also handle the `next` branch and pre-releases (or we could copy it and make a pre-release version).

We are not sure what is best to do that the moment, but we want to have the pre-releases handled by Github Actions, so we are investigating how to do so.
